src/main/java/org/terasology/Lakes/LakeProvider.java

```diff
@@ -61,36 +61,54 @@ public void process(GeneratingRegion region) {
         Region3i worldRegion = lakes.getWorldRegion();
         BiomeFacet biomeFacet = region.getRegionFacet(BiomeFacet.class);
 
-        for (Vector3i pos : worldRegion) {
-            float sHeight = surfaceHeightFacet.getWorld(pos.x(), pos.z());
-            float probability;
-            probability = computeProbability(pos, biomeFacet);
-            if (pos.y() < sHeight - 20 && probability > 0.999999) {
-                lakes.add(new Lake(pos, 10 + Math.round(20 * Math.abs(noise.noise(pos.x(), pos.z())))));
-            } else if (pos.y() == Math.round(sHeight) && probability > 0.9999 && checkGradient(pos,
-                    surfaceHeightFacet)) {
-                Lake temp = new Lake(pos, 10 + Math.round(20 * Math.abs((noise.noise(pos.x(), pos.z())))));
-                if (checkCorners(temp.getBoundingBox(), surfaceHeightFacet)) {
-                    int minHeight = getMinimumHeight(temp.getBoundingBox(), surfaceHeightFacet);
-                    if (minHeight < pos.y()) {
-                        temp.setWaterHeight(minHeight);
+        Vector3i min = worldRegion.min();
+        int step = 3;
```

Comment:
This one could perhaps also use a quick comment? It doesn't seem to change, so maybe a well-named constant would be even better